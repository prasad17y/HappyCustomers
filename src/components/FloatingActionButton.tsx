import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  Text,
  View,
  ViewStyle,
  StyleProp,
} from 'react-native';

interface FloatingActionButtonProps {
  onPress: () => void;
  label: string;
  style?: StyleProp<ViewStyle>;
}

const FloatingActionButton: React.FC<FloatingActionButtonProps> = ({
  onPress,
  label,
  style,
}) => {
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      <View style={styles.button}>
        <Text style={styles.plus}>{label}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: 60,
    height: 60,
    borderRadius: 8,
    backgroundColor: '#0b5ac2',
    justifyContent: 'center',
    alignItems: 'center',
  },
  plus: {
    color: '#fff',
    fontSize: 30,
    lineHeight: 32,
  },
});

export default React.memo(FloatingActionButton);
