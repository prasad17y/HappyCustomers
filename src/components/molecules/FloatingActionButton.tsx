import React from 'react';
import {StyleSheet, Text} from 'react-native';
import Button from '../atoms/Button';

interface FloatingActionButtonProps {
  onPress: () => void;
}

const FloatingActionButton: React.FC<FloatingActionButtonProps> = ({
  onPress,
}) => {
  return (
    <Button style={styles.container} onPress={onPress}>
      <Text style={styles.plus}>+</Text>
    </Button>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 30,
    right: 30,
    width: 60,
    height: 60,
    borderRadius: 30,
    backgroundColor: '#0b5ac2',
    elevation: 8,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 4},
    shadowOpacity: 0.3,
    shadowRadius: 4,
  },
  plus: {
    color: '#fff',
    fontSize: 30,
    lineHeight: 32,
  },
});

export default React.memo(FloatingActionButton);
