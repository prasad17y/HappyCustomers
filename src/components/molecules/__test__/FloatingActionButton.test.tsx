import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import FloatingActionButton from '../FloatingActionButton';

describe('FloatingActionButton', () => {
  it('should render the button with a plus icon', () => {
    render(<FloatingActionButton onPress={jest.fn()} />);
    expect(screen.getByText('+')).toBeTruthy();
  });

  it('should call onPress when the button is pressed', () => {
    const onPressMock = jest.fn();
    render(<FloatingActionButton onPress={onPressMock} />);

    const button = screen.getByText('+');
    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('should match the snapshot', () => {
    const tree = render(<FloatingActionButton onPress={jest.fn()} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
