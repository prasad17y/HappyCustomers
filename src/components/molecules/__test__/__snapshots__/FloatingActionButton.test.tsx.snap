// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingActionButton should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#0b5ac2",
      "borderRadius": 30,
      "bottom": 30,
      "elevation": 8,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "right": 30,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4,
      "width": 60,
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 30,
        "lineHeight": 32,
      }
    }
  >
    +
  </Text>
</View>
`;
